openid4vc-high-assurance-interoperability-profile-1_0.md

```diff
@@ -231,7 +231,9 @@ The following requirements apply for both, the Wallet and the Verifier, unless s
   * The Wallet MUST support both signed and unsigned requests as defined in Annex A.3.1 and A.3.2 of [@!OIDF.OID4VP]. The Verifier MAY support signed requests, unsigned requests, or both.
 * Wallet Invocation is done via the W3C Digital Credentials API or an equivalent platform API. Any other mechanism, including Custom URL schemes, MUST NOT be used.
 * Response Mode MUST be `dc_api.jwt`. The response MUST be encrypted.
-* Specific requirements for the response encryption are tbd (https://github.com/openid/oid4vc-haip/issues/131).
+* Response encryption MUST be performed as specified in [@!OIDF.OID4VP, section 7.3]. The JWE `alg` (algorithm) header parameter (see [@!RFC7516, section 4.1.1])
+  value MUST be `ECDH-ES` (as defined in [@!RFC7518, section 4.6]), with key agreement utilizing keys on the `P-256` curve (see [@!RFC7518, section 6.2.1.1]).
+  The JWE `enc` (encryption algorithm) header parameter (see [@!RFC7516, section 4.1.2]) value MUST be `A128GCM` (as defined in [@!RFC7518, section 5.3]).
```

Comment:
we might want to put it in a separate algorithms section later on, but this should be a good start

Reply:
Yeah, I was trying to do only what had been decided on about encryption by the WG to meet the stated requirements from ISO. I am not convinced this is a good idea or it actually fulfills the requirements.  Getting to "all the details required to have interoperability need to be specified" involves more than just some encryption algs. And something like an MTI set of algs across the thole ting rather than mandating specific ones for one piece would probably be much better. But after circling around and around the HPKE discussions and being told that JARM and JWE were not implementable or interoperable, I am trying to move things forward with this as per the WG decision as I understood it to be.

Reply:
I think the slight adjustment in https://github.com/openid/oid4vc-haip/pull/155#discussion_r1918489096 would be much better. WDYT?

Reply:
yes, agreed.